### PR TITLE
Use os_sdn network on compute nodes when using VXLAN

### PR DESCRIPTION
**Pull request by vuntz**

### CODE REVIEW

File: crowbar_framework/app/models/nova_service.rb [hunk 1]
```diff
@@ -188,10 +188,11 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
       # TODO(toabctl): The same code is in the neutron barclamp. Should be extracted and reused!
       #                (see crowbar_framework/app/models/neutron_service.rb)
       if neutron["attributes"]["neutron"]["networking_plugin"] == "ml2"
-        if neutron["attributes"]["neutron"]["ml2_type_drivers"].include?("gre")
+        ml2_type_drivers = neutron["attributes"]["neutron"]["ml2_type_drivers"]
+        if ml2_type_drivers.include?("gre") || ml2_type_drivers.include?("vxlan")
```
Comment (aspiers) [inline crowbar_framework/app/models/nova_service.rb]: Could use same [`&` trick](https://github.com/crowbar/barclamp-neutron/pull/206#discussion_r26844719) here too.

**Comment by aspiers** [comment]
LGTM apart from nitpick.

**Comment by nkrinner** [comment]
+1

File: crowbar_framework/app/models/nova_service.rb (hunk 1 above)
Comment (vuntz) [inline crowbar_framework/app/models/nova_service.rb]: Let's wait for the conclusion in the other review for this one :-)

Comment (tboerger) [inline crowbar_framework/app/models/nova_service.rb]: I dislike this, the current code is better readable especially as many people don't know what the ```&``` operator does in this case.

**Comment by tboerger** [comment]
:+1:

**Pull request merged**